web/html/src/manager/clusters/cluster/cluster-overview.js

```diff
@@ -0,0 +1,328 @@
+  const [selections, setSelections] = useState<Set<string>>(new Set());
+  const [nodes, setNodes] = useState<Array<ClusterNodeType>>([]);
+  const [nodeDetailFields, setNodeDetailFields] = useState<Array<string>>([]);
+  const [fetching, setFetching] = useState<boolean>(false);
+  const [fetchingProps, setFetchingProps] = useState<boolean>(false);  
+  const [editModel, setEditModel] = useState<EditableClusterPropsType>({name: "", description: ""});
+  const nodesForm = useRef(null);
+
+  const {fetchClusterNodes, refreshGroupNodes, saveClusterProps, fetchClusterProps} = useClustersApi();
+
+  const fetchData = () => {
+    setFetching(true);
+    fetchClusterNodes(props.cluster.id)
+    .then((clusterNodes) => {
+      setNodes(clusterNodes.nodes);
+      setNodeDetailFields(clusterNodes.fields);
+      setSelections(new Set());
+    })
+    .catch((error : ErrorMessagesType) => {
+      props.setMessages(error.messages);
+    })
+    .finally((fetching) => setFetching(false));
+  }
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  const onRemove = () => {
+    if (selections.size > 0 && nodesForm.current) {
+      nodesForm.current.submit();
+    }
+  };
+
+  const onRefreshGroupNodes = () => {
+    refreshGroupNodes(props.cluster.id)
+    .then((actionId) => props.setMessages([Messages.success(t("System group refresh action scheduled successfully"))]))
+    .catch((error : ErrorMessagesType) => {
+      props.setMessages(error.messages);
+    })
+  };
+
+  const onFetchClusterProps = () => {
+    setFetchingProps(true);
+    fetchClusterProps(props.cluster.id)
+      .then((data: ClusterType) => {
+        setCluster(data);
+        props.onUpdateName(data.name);
+      })
+      .catch((error : ErrorMessagesType) => {
+        props.setMessages(error.messages);
+      })      
+      .finally(() => {
+          setFetchingProps(false);
+      });    
+  }
+
+  const onSaveClusterProps = () => {
+    return saveClusterProps(props.cluster.id, editModel)
+      .then(_ => {
+        closeDialog("cluster-edit-dialog");
+        props.setMessages([Messages.success(t("Cluster properties updated successfully"))]);
+        onFetchClusterProps();
+      })
+      .catch((error : ErrorMessagesType) => {
+        props.setMessages(error.messages);
+      });
+  };
+
+  const filterFunc = (row, criteria) => {
+      const keysToSearch = ['hostname'];
+      if (criteria) {
+          return keysToSearch.map(key => row[key]).join().toLowerCase().includes(criteria.toLowerCase());
+      }
+      return true;
+  };
+
+  const onSelectNode = (event, hostname) => {
+      const newSelections = new Set(selections);
+      if (event.target.checked) {
+          newSelections.add(hostname);
+      } else {
+          newSelections.delete(hostname);
+      }
+      setSelections(newSelections);
+  }  
+
+  const editContent =
+      <Form
+          model={editModel}
+          onValidate={(valid) => console.log("valid: " + (valid ? "true" : "false"))}
```

Comment:
I guess this is leftover from debugging.
```suggestion
          onValidate={setValid}
```
Where `setValid` is the setter of an `isValid` flag:
```js
const [isValid, setValid] = useState<boolean>(true);
```
Then saving should be prevented/button disabled according to the flag.

Reply:
done